packages/codemirror-blocks/spec/api-test.ts

```diff
@@ -425,7 +425,9 @@ describe("when testing CM apis,", () => {
     );
   });
 
-  it("setCursor should work as-is for text, or activate the containing block", async () => {
+  // TODO(pcardune): reenable or rewrite this test, which was passing by accident
+  // due to activateByNid being async and not waiting for it.
+  xit("setCursor should work as-is for text, or activate the containing block", async () => {
     cmb.setBlockMode(true);
     await finishRender();
     cmb.setCursor({ line: 1, ch: 1 });
```

Comment:
Now that the call to `setFocusedNode()` happens synchronously, this test (and only this test) started to fail. If I were to leave the call to `setFocusedNode()` where it was, but add an `await finishRender()` after the call to `setCursor()` in this test, then the test would also fail. This implies that the test was passing because it was looking at state before it changed.

I tried debugging this and figuring out what the behavior the test was actually trying to test but I can't quite figure it out.

Reply:
The idea is that calling `editor.setCursor({line: 1, ch: 1})` should put the cursor exactly _there_ in text-mode, but should instead activate the containing block in block-mode.

Reply:
ok, so the first line of the tests, which puts the editor into blockMode, actually shouldn't do that?

Reply:
Whoa. Correct.